pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -171,25 +174,32 @@ func (c *Cluster) checkHealth() error {
 		}
 
 		logger.Debugf("mon %q NOT found in quorum. Mon quorum status: %+v", mon.Name, quorumStatus)
-		allMonsInQuorum = false
+		// if the time out is set to 0 this indicate that we don't want to trigger mon failover
+		if MonOutTimeout == time.Duration(0) {
+			logger.Warningf("mon %q NOT found in quorum and health timeout is 0, mon will never fail over", mon.Name)
+			return nil
+		} else {
```

Comment:
Since there is a `return` on the previous line, you don't need the `else` statement, right? Then we can keep the indentation as before.